File: plugins/za-qe-tools/scripts/auto-setup.js
#!/usr/bin/env node
/**
 * SessionStart hook: za-qe-tools 自动初始化
 *
 * 逻辑：
 * 1. 确保 ~/.claude/za-qe-tools.json 配置文件存在
 * 2. 根据 statusline 配置自动写入 settings.json
 * 3. 如果 dippy 已启用，确保 ~/.dippy/config 存在
 *
 * @author Alfie
 */
'use strict';

const fs   = require('fs');
const path = require('path');
const os   = require('os');

const SETTINGS_PATH = path.join(os.homedir(), '.claude', 'settings.json');
const CONFIG_PATH   = path.join(os.homedir(), '.claude', 'za-qe-tools.json');
const DIPPY_CONFIG  = path.join(os.homedir(), '.dippy', 'config');

const PLUGIN_ROOT = process.env.CLAUDE_PLUGIN_ROOT || '';

const DEFAULT_CONFIG = {
  statusline: { enabled: true, mode: 'powerline' },
  notify:     { enabled: false },
  dippy:      { enabled: false },
  esp:        { enabled: false }
};

function readJson(filePath) {
  try {
    return JSON.parse(fs.readFileSync(filePath, 'utf8'));
  } catch (_) {
    return null;
  }
}

function ensureModuleConfig() {
  if (fs.existsSync(CONFIG_PATH)) return readJson(CONFIG_PATH);

  // 首次启动，创建默认配置
  try {
    fs.mkdirSync(path.dirname(CONFIG_PATH), { recursive: true });
    fs.writeFileSync(CONFIG_PATH, JSON.stringify(DEFAULT_CONFIG, null, 2), 'utf8');
  } catch (_) {}

  return DEFAULT_CONFIG;
}

function setupStatusline(config, settings) {
  if (!PLUGIN_ROOT) return settings;

  const slConfig = config.statusline || {};
  if (!slConfig.enabled) {
    // statusline 关闭时不主动删除已有配置（用户可能手动配置了）
    return settings;
  }

  if (settings.statusLine) {
    // 已配置，跳过
    return settings;
  }

  // 未配置，根据 mode 自动写入
  const scriptName = slConfig.mode === 'standard' ? 'statusline.py' : 'statusline-powerline.py';
  const scriptPath = PLUGIN_ROOT.replace(/\\/g, '/') + '/scripts/' + scriptName;

  settings.statusLine = {
    type: 'command',
    command: `uv run ${scriptPath}`,
    padding: 2
  };

  return settings;
}

function setupDippy(config) {
  if (!config.dippy || !config.dippy.enabled) return;
  if (!PLUGIN_ROOT) return;

  // 确保 ~/.dippy/config 存在
  if (fs.existsSync(DIPPY_CONFIG)) return;

  const defaultDippy = path.join(PLUGIN_ROOT, 'config', 'default.dippy');
  if (!fs.existsSync(defaultDippy)) return;

  try {
    fs.mkdirSync(path.dirname(DIPPY_CONFIG), { recursive: true });
    fs.copyFileSync(defaultDippy, DIPPY_CONFIG);
  } catch (_) {}
}

function main() {
  const config = ensureModuleConfig();

  let settings = readJson(SETTINGS_PATH);
  if (!settings) settings = {};

  const originalSettings = JSON.stringify(settings);
  settings = setupStatusline(config, settings);

  // 仅在配置有变化时写入
  if (JSON.stringify(settings) !== originalSettings) {
    try {
      fs.mkdirSync(path.dirname(SETTINGS_PATH), { recursive: true });
      fs.writeFileSync(SETTINGS_PATH, JSON.stringify(settings, null, 2), 'utf8');
    } catch (_) {}
  }

  setupDippy(config);

  process.stdout.write(JSON.stringify({ suppressOutput: true }) + '\n');
}

main();

File: plugins/za-qe-tools/scripts/auto-setup.py
# /// script
# dependencies = []
# ///
"""
StatusLine 配置脚本。

用法:
  SessionStart 自动调用（无参数）：首次写入 powerline，已有则修正路径
  config 命令调用（带参数）：切换模式
    uv run auto-setup.py powerline
    uv run auto-setup.py standard
    uv run auto-setup.py off
"""
from __future__ import annotations

import json
import os
import sys
from pathlib import Path

SETTINGS_PATH = Path.home() / ".claude" / "settings.json"
PLUGIN_ROOT = os.environ.get("CLAUDE_PLUGIN_ROOT", "")

SCRIPT_MAP = {
    "powerline": "statusline-powerline.py",
    "standard": "statusline.py",
}


def read_json(file_path: Path) -> dict | None:
    try:
        return json.loads(file_path.read_text(encoding="utf-8"))
    except (OSError, json.JSONDecodeError):
        return None


def write_json(file_path: Path, data: dict) -> None:
    try:
        file_path.parent.mkdir(parents=True, exist_ok=True)
        file_path.write_text(json.dumps(data, indent=2), encoding="utf-8")
    except OSError:
        pass


def make_statusline_config(mode: str) -> dict:
    script_name = SCRIPT_MAP[mode]
    script_path = PLUGIN_ROOT.replace("\\", "/") + "/scripts/" + script_name
    return {
        "type": "command",
        "command": f"uv run {script_path}",
        "padding": 2,
    }


def main() -> None:
    if not PLUGIN_ROOT:
        print(json.dumps({"suppressOutput": True}))
        return

    settings = read_json(SETTINGS_PATH) or {}
    original = json.dumps(settings)

    args = sys.argv[1:]
    mode = args[0] if args else None

    if mode == "off":
        # 删除 statusLine
        settings.pop("statusLine", None)
    elif mode in SCRIPT_MAP:
        # 指定模式 → 写入
        settings["statusLine"] = make_statusline_config(mode)
    else:
        # 无参数（SessionStart 自动调用）
        if "statusLine" not in settings:
            # 首次：默认 powerline
            settings["statusLine"] = make_statusline_config("powerline")
        else:
            # 已有：检查路径是否指向当前 PLUGIN_ROOT，过期则修正
            current_cmd = settings.get("statusLine", {}).get("command", "")
            if PLUGIN_ROOT.replace("\\", "/") not in current_cmd and "statusline" in current_cmd.lower():
                # 路径过期，检测当前模式并修正
                if "statusline-powerline" in current_cmd:
                    settings["statusLine"] = make_statusline_config("powerline")
                elif "statusline.py" in current_cmd:
                    settings["statusLine"] = make_statusline_config("standard")

    if json.dumps(settings) != original:
        write_json(SETTINGS_PATH, settings)

    print(json.dumps({"suppressOutput": True}))


if __name__ == "__main__":
    main()
